src/Tribe/Shortcodes/User_Event_Confirmation_List.php

```diff
@@ -80,102 +91,87 @@ protected function generate_attendance_list() {
-		$event_keys = array();
-
-		foreach ( Tribe__Tickets__Tickets::modules() as $module_class => $module_instance ) {
-			/**
-			 * The usage of plain `$module_class::ATTENDEE_EVENT_KEY` will throw a `T_PAAMAYIM_NEKUDOTAYIM`
-			 * when using PHP 5.2, which is a fatal.
-			 *
-			 * So we have to construct the constant name using a string and use the `constant` function.
-			 */
-			if ( defined( "$module_class::ATTENDEE_EVENT_KEY" ) ) {
-				$event_keys[] = constant( "$module_class::ATTENDEE_EVENT_KEY" );
-			} else {
-				$event_keys[] = call_user_func( array( $module_class, 'get_key' ), 'ATTENDEE_EVENT_KEY' );
+		$post_ids                    = [];
+		$ticket_ids                  = [];
+		$event_ids                   = [];
+		$upcoming_non_event_post_ids = [];
+
+		// Get all of this user's attendee records
+
+		/** @var Tribe__Tickets__Attendee_Repository $attendees */
+		$attendees = tribe_attendees();
+
+		$attendee_ids = $attendees
+			->by( 'user', $this->params['user'] )
+			->fields( 'ids' )
+			->get_ids();
+
+		// Build list of Posts and Tickets this user's attendance applies to
+
+		/** @var Tribe__Tickets__Tickets_Handler $tickets_handler */
+		$tickets_handler = tribe( 'tickets.handler' );
+
+		foreach ( $attendee_ids as $attendee_id ) {
+			$connection = $tickets_handler->get_object_connections( $attendee_id );
+
+			if ( ! empty( $connection->event ) ) {
+				$post_ids[] = $connection->event;
+			}
+
+			if ( ! empty( $connection->product ) ) {
+				$ticket_ids[] = $connection->product;
 			}
 		}
 
-		return $event_keys;
-	}
+		$post_ids = array_unique( $post_ids );
 
-	/**
-	 * Provides a quoted, comma separated and escaped list of meta keys used to link
-	 * attendee posts to event posts.
-	 *
-	 * @return string
-	 */
-	protected function build_escaped_key_list( array $keys ) {
-		global $wpdb;
+		// Restrict to upcoming Tribe Events posts
+		if ( function_exists( 'tribe_events' ) ) {
+			/** @var Tribe__Events__Repositories__Event $events */
+			$events = tribe_events();
 
-		foreach ( $keys as &$key ) {
-			$key = $wpdb->prepare( '%s', $key );
+			$event_ids = $events
+				->by( 'ends_after', 'now' )
+				->in( $post_ids )
+				->fields( 'ids' )
+				->get_ids();
 		}
 
-		return implode( ',', $keys );
+		// Restrict non-Tribe Events posts to those for which ticket availability end date has not ended (regardless of available capacity)
+		$ticket_ids = array_unique( $ticket_ids );
+
+		/** @var Tribe__Tickets__Ticket_Repository $tickets */
+		$tickets = tribe_tickets();
+
+		$ticket_ids_upcoming = $tickets
+			->in( $ticket_ids )
+			->by( 'available_until', 'now' )
+			->fields( 'ids' )
+			->get_ids();
+
+		foreach ( $ticket_ids_upcoming as $ticket_id_upcoming ) {
+			$connection = $tickets_handler->get_object_connections( $ticket_id_upcoming );
+
+			if ( ! empty( $connection->event ) ) {
+				$upcoming_non_event_post_ids[] = $connection->event;
+			}
+		}
+
+		// Get list of Tribe Events posts plus Non-Tribe Events posts
```

Comment:
:no_entry_sign: Inline comments must end in full-stops, exclamation marks, or question marks

<sub>hash: b38062e49b2a20caed40369fb5118f3f</sub>